src/Adapter/Product/QueryHandler/SearchProductsHandler.php

```diff
@@ -300,6 +300,16 @@ private function getProductPriceForOrder(
             return Product::getPriceStatic($productId, $withTaxes, $productAttributeId, $computingPrecision);
         }
 
+        // If the product already exists in the order, whatever his price, we take the same value
+        foreach ($order->getCartProducts() as $cartProduct) {
+            if (
+                (int) $cartProduct['id_product'] === $productId
+                && (int) $cartProduct['id_product_attribute'] === $productAttributeId
```

Comment:
Maybe check the customization too?

Reply:
Does the custimation impact the product's price ?